db/migrations_tests/deals_fast_retrieval_test.go

```diff
@@ -22,11 +22,15 @@ func TestDealFastRetrieval(t *testing.T) {
 	req.NoError(goose.SetDialect("sqlite3"))
 	req.NoError(goose.UpTo(sqldb, ".", 20221124191002))
 
-	// Generate 2 deals
+	// Generate 1 deal
 	dealsDB := db.NewDealsDB(sqldb)
 	deals, err := db.GenerateNDeals(1)
 	req.NoError(err)
 
+	// Add AnnounceToIPNI to allow tests to works
+	_, err = sqldb.Exec(`ALTER TABLE Deals ADD AnnounceToIPNI BOOL`)
+	require.NoError(t, err)
+
 	// Insert the deals in DB
 	err = dealsDB.Insert(ctx, &deals[0])
 	require.NoError(t, err)
```

Comment:
I used the Insert column method to avoid writing a very lengthy manual DB INSERT. Also, if we manually add the entry to DB, we are bypassing all the deal insert code. This method seemed the most clean to me.
Can you please let me know why manual INSERT makes more sense?

Reply:
At the beginning of the test we run migrations up to a certain point. But the database management code is written to work with the latest database structure. So if we use the `dealsDB.Insert(...)` method it will fail because the code references fields that do not exist in the database structure until all migrations have been run.

One way around that is to write an ALTER TABLE statement as you have done. However in that case you are effectively doing the same thing as running all the database migrations. Each time we add a new migration you will need to add more code to each of these migration tests to manually perform the migration.

So instead I think we should write a manual INSERT statement that works with the structure of the database at the point the migration has reached.

Reply:
That makes sense. I will update both tests using the ALTER statement.